task/host-sp-comms/src/main.rs

```diff
@@ -853,15 +853,15 @@ cfg_if::cfg_if! {
 }
 
 fn sp_to_sp3_interrupt_enable(sys: &sys_api::Sys) {
-    sys.gpio_set(SP_TO_SP3_INT_L).unwrap();
-
     sys.gpio_configure_output(
         SP_TO_SP3_INT_L,
-        sys_api::OutputType::PushPull,
+        sys_api::OutputType::OpenDrain,
         sys_api::Speed::Low,
-        sys_api::Pull::Down,
+        sys_api::Pull::None,
     )
     .unwrap();
+
+    sys.gpio_set(SP_TO_SP3_INT_L).unwrap();
```

Comment:
Why do we move the `gpio_set` to after the configure call? (FWIW I originally had it this way, but @mkeeter corrected me https://github.com/oxidecomputer/hubris/pull/861#discussion_r999810734)

Reply:
We never want to actively drive this pin high from the SP end and I presumed that setting it before configuring as open drain would cause the SP to do that. I'll defer to @mkeeter here.

Reply:
I think the `gpio_set` call should be called **before** `gpio_configure_output`.  @citrus-it the missing bit of info is that pins come out of reset in analog mode, rather than as outputs.

Here's what happens when `gpio_set` is called after `gpio_configure_output` (the bad order):

- When the SP comes out of reset, this pin is configured in analog mode, meaning the pin is High-Z (and therefore pulled up by the external resistor).
- Calling `gpio_configure_output` switches to output, open-drain mode.  Because the output data register `ODRx` is be 0 (its default value), the output driver will enable the N-MOS, pulling the pin down to ground.
- Then, the `gpio_set` call will set `ODRx` to 1, the output driver will go High-Z, and we'll be pulled back up by the external resistor.

The net effect will be a High → Low → High glitch.

If you call `gpio_set` beforehand (the good order):

- When the SP comes out of reset, this pin is configured in analog mode, meaning the pin is High-Z (and therefore pulled up by the external resistor).
- Calling `gpio_set` will set `ODRx` to 1, which is ignored in analog mode
- Calling `gpio_configure_output` switches us to open-drain mode; because `ODRx` is already high, we stay High-Z and there is no glitch.

One corner case: if the task restarts, then the pin is _already_ configured in open-drain mode, which is also fine.

Reply:
Thanks for the detail, I've put the order back at it was, rebased and pushed.